HACKING.md

```diff
@@ -396,6 +412,148 @@ multipass delete juju
+
+First, create a new Multipass container. You may need to adjust the resources
+depending on your host machine, but you will need to allocate at least 20GB of
+disk space.
+
+```shell
+multipass launch --cpus 2 --disk 20G --memory 8G --name jimm
+```
+
+Copy your ssh key into the container. You can do this manually or use this one-liner:
+
+```shell
+cat ~/.ssh/id_rsa.pub | multipass exec jimm -- tee -a .ssh/authorized_keys
+```
+
+SSH into the container:
+
+```shell
+ssh -A ubuntu@[multipass.ip]
+```
+
+#### Set up JIMM
+
+Check out the JIMM repository:
+
+```shell
+git clone git@github.com:canonical/jimm.git
+cd jimm
+```
+
+Start by making a small configuration change so that the login process redirects
+to the development dashboard URL:
+
+```shell
+nano docker-compose.yaml
+```
+
+Find `JIMM_DASHBOARD_FINAL_REDIRECT_URL` and set it to `"http://jimm.localhost:8036"`.
+
+Next follow the steps in the [Starting the
+environment](https://github.com/canonical/jimm/tree/v3/local#starting-the-environment)
+section of the JIMM docs.
+
+Once the environment is running it will be steadily
+outputting openfga health checks like the following, at which
+point you can move on to the next steps:
+
+```shell
+openfga                      | 2024-06-03T02:25:47.168Z	INFO	grpc_req_complete	{"grpc_service": "grpc.health.v1.Health", "grpc_method": "Check", "grpc_type": "unary", "request_id": "db6a8859-f296-44b3-b150-a1dff5be93fe", "raw_request": {"service":""}, "raw_response": {"status":"SERVING"}, "peer.address": "127.0.0.1:49320", "grpc_code": 0}
+```
+
+In a new terminal, enter the container, this time following the [port forward
+instructions](#forward-ports) and then go to the JIMM directory.
+```shell
+cd jimm
+```
+
+Now follow the [Q/A Using
+jimmctl](https://github.com/canonical/jimm/tree/v3/local#qa-using-jimmctl)
+steps.
+
+#### Forward ports
+
+To expose the various JIMM APIs so that they can be accessed from outside of the
+Multipass container, you can use an SSH port forward. This will also enable
+access to the Multipass using the hostnames set up inside the Multipass e.g. jimm.localhost.
+
+From your host machine run the following:
+
+```shell
+export JIMM_CONTAINER=[multipass.ip]
+ssh -A -L :8082:$JIMM_CONTAINER:8082 -L :17070:$JIMM_CONTAINER:17070  -L :443:$JIMM_CONTAINER:443 -L :8036:$JIMM_CONTAINER:8036 ubuntu@$JIMM_CONTAINER
+```
+
+#### Set up Juju Dashboard
+
+Start by getting your fork of the dashboard:
+
+```shell
+cd ~
+git clone git@github.com:[your-username]/juju-dashboard.git
+cd juju-dashboard
+```
+
+Install Node.js an Yarn:
+
+```shell
+sudo snap install node --classic
+```
+
+Install the dependencies
+
+```shell
+yarn install
+```
+
+Copy the configuration file:
+```shell
+cp public/config.js public/config.local.js
+```
+
+Change the configuration as follows:
+
+```shell
+nano public/config.local.js
+// Change these options:
+controllerAPIEndpoint: "ws://jimm.localhost:17070/api",
+identityProviderAvailable: false,
+```
```

Comment:
I think it might be easier to copy-paste the code if it is split into 2 chunks: one only with `nano public/config.local.js` and the other one detailing the changes needed to be performed.

Reply:
Done.